### ISSUE TITLE
Can't create user without admin account (v15)

### ISSUE THREAD

**Issue by Hortiks** [Dolibarr/dolibarr#30465]
### Bug

There is a problem with the creation of users using the api of Dolibarr v15 (Code error 401 with no admin account, BUT with permissions)
The problem comes from the function that checks users authorizations on api_users.class.php.

The function is
`// Check user authorization
		if (empty(DolibarrApiAccess::$user->rights->user->creer) && empty(DolibarrApiAccess::$user->admin)) {
			throw new RestException(401, "User creation not allowed for login ".DolibarrApiAccess::$user->login);
		}`
		
and should be
`// Check user authorization
		if (empty(DolibarrApiAccess::$user->rights->user->user->creer) && empty(DolibarrApiAccess::$user->admin)) {
			throw new RestException(401, "User creation not allowed for login ".DolibarrApiAccess::$user->login);
		}`
		
Thanks to TC_L from the forum Dolibarr that found the solution :
link to the french topic : [https://www.dolibarr.fr/forum/t/pourquoi-je-ne-peux-pas-creer-dutilisateur-malgre-les-droits/46995](url)

### Dolibarr Version

v15.0.0

### Environment PHP

7.3.12

### Environment Database

Mariadb 10.4.10

### Steps to reproduce the behavior and expected behavior

Try to create a new user using api with a non admin account.
Even if your non admin account has permissions, you'll get Code error 401.

**Comment by williamdes** [comment]
Fixed in 78f90982c475e6cd2d8a1cb94f7e5b7911da456f by #36337

I ran `sed -i "s/hasRight('user', 'creer')/hasRight('user', 'user', 'creer')/" user/class/api_users.class.php` in Dolibarr+Docker v22.0.3

**Comment by williamdes** [comment]
@eldy what is the LTS policy  ?
If I read https://wiki.dolibarr.org/index.php/Versions correctly it would be v18
Should I backport the permissions fix back to the v18 branch so you merge it up to v22 ?

**Comment by williamdes** [comment]
Hello @rycks 

Here is the issue that can be backported
See my comment above for the commit to cherry-pick: https://github.com/Dolibarr/dolibarr/issues/30465#issuecomment-3794772888

**Comment by rycks** [comment]
@williamdes yes if you make a PR on dolibarr 18.0 then this one will be merged into 19....21

**Comment by williamdes** [comment]
Bug introduced by v14 in https://github.com/Dolibarr/dolibarr/commit/3c9d8bc931799f9cbf7e04d6cedcf1a3bff44ab8
I did open a PR for v14: https://github.com/Dolibarr/dolibarr/pull/37502 cc @eldy it reverts your change mentioned above

Since V20 the syntax changes (because of https://github.com/Dolibarr/dolibarr/commit/7b54824d49f53df651860ae91e768d64170e5c6a#diff-64e82e2f84fc5a1af766e17f0e533221c48231c6bbe3f342af899ee6854748fb):
- https://github.com/Dolibarr/dolibarr/blob/584e103ebf0cbc4013314d389d233ecf60acf636/htdocs/user/class/api_users.class.php#L309

I made a PR for v20: https://github.com/Dolibarr/dolibarr/pull/37504

This should be all that is needed, you can merge https://github.com/Dolibarr/dolibarr/pull/37502 until v20
And https://github.com/Dolibarr/dolibarr/pull/37504 into v20 and above
Version v23 is fixed by https://github.com/Dolibarr/dolibarr/pull/36337
All versions after are normally good
